### DISCUSSION TITLE
Reading 125 consecutive holding registers with Serial ASCII with one request

### DISCUSSION THREAD

**Discussion by maovidal** [pymodbus-dev/pymodbus#2294] [Help] [answered]
Hi:

Prior to **v3.7.0** I was able to read 125 consecutive holding registers with one request using a client configured for **Serial ASCII** and using the **read_holding_registers()** function.

I have found that the reason I'm now getting an error indicating that 250 bytes are received is the limit introduced with #2260 to the register_read_message here: [on the file pymodbus/pdu/register_read_message.py ](https://github.com/pymodbus-dev/pymodbus/blob/3e60a20cef282fed59281cded7551b9f231a7786/pymodbus/pdu/register_read_message.py#L92)

From what I understand, reading the section **2.5.2.1 MODBUS Message ASCII Framing** of the [MODBUS over serial line specification and implementation guide V1.02](https://modbus.org/docs/Modbus_over_serial_line_V1_02.pdf), up to 252 chars are available for data. 

Now, from the section **6.3 03 (0x03) Read Holding Registers** from the [MODBUS Application Protocol Specification V1.1b3](https://modbus.org/docs/Modbus_Application_Protocol_V1_1b3.pdf) the Response would need two chars for the byte count and 250 for the registers value. Then those 252 chars would be in the limit of the serial line implementation.

Should be the limit allow those 125 registers per the spec? Or Am I missing something?

Thank you for any help on this!

**Comment by janiversen** [comment]
I think this is a bug, I will investigate.

**Comment by janiversen** [comment, marked as answer]
Solved in #2295, thanks for reporting this bug.
